src/compression.jl

```diff
@@ -75,21 +114,32 @@ end
 
 function TranscodingStreams.process(codec::ZstdCompressor, input::Memory, output::Memory, error::Error)
     cstream = codec.cstream
-    cstream.ibuffer.src = input.ptr
-    cstream.ibuffer.size = input.size
-    cstream.ibuffer.pos = 0
+    ibuffer_starting_pos = UInt(0)
+    if codec.endOp == LibZstd.ZSTD_e_end &&
+       cstream.ibuffer.size != cstream.ibuffer.pos
+            # While saving a frame, the prior process run did not complete processing.
+            # Re-run with the same size and pos
+            ibuffer_starting_pos = cstream.ibuffer.pos
+            # Set the pointer relative to input.ptr, but keep size and pos
+            cstream.ibuffer.src = input.ptr - cstream.ibuffer.pos
```

Comment:
This may read out of bounds of `input` if `cstream.ibuffer.pos` is greater than zero.

Reply:
Why is that? The idea is for `cstream.ibuffer.src + cstream.ibuffer.pos == input.ptr`. We want to start reading at `input.ptr` while keeping `cstream.ibuffer.pos` and `cstream.ibuffer.size` as positive values.

Reply:
Okay, that makes sense. If `ibuffer` gets reset in `startproc` I can't think of a way for this to go wrong.